src/avtk/clipselector.cxx

```diff
@@ -264,18 +351,28 @@ int ClipSelector::handle(int event)
 				}
 
 
+                bool empty =  clips[clipNum].getState() == GridLogic::STATE_EMPTY;
 				Fl_Menu_Item rclick_menu[] = {
 					{ "Load" },
 					{ "Save" },
 					{ "Special"},
 					{ "Beats",  0,   0, 0, FL_SUBMENU | FL_MENU_DIVIDER },
-					{"1       "},
-					{"2"},
-					{"4"},
-					{"8"},
-					{"16"},
-					{"32"},
-					{"64"},
+					RECORD_LENGTH_MENU_ITEM(1),
+					RECORD_LENGTH_MENU_ITEM(2),
+					RECORD_LENGTH_MENU_ITEM(4),
+					RECORD_LENGTH_MENU_ITEM(8),
+					RECORD_LENGTH_MENU_ITEM(16),
+					RECORD_LENGTH_MENU_ITEM(32),
+					RECORD_LENGTH_MENU_ITEM(64),
+					{0},
+					{ "Bars to record",  0,   0, 0, FL_SUBMENU | FL_MENU_DIVIDER},
+					RECORD_BARS_MENU_ITEM(1),
+					RECORD_BARS_MENU_ITEM(2),
+					RECORD_BARS_MENU_ITEM(4),
+					RECORD_BARS_MENU_ITEM(6),
+					RECORD_BARS_MENU_ITEM(8),
+					{"Endless", 0, setRecordBarsCb, (void*)-1, FL_MENU_DIVIDER | FL_MENU_RADIO | ((clips[clipNum].getBeatsToRecord() <= 0) ? FL_MENU_VALUE : 0) | (empty ? 0 : FL_MENU_INACTIVE)},
+					{"Custom", 0, setRecordBarsCb, (void*)-2, empty ? 0 : FL_MENU_INACTIVE},
```

Comment:
@harryhaaren Do you think passing `-2` as content is a problem here? Seems to be the same like in the other menu entries...